src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -744,8 +749,12 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return newKernelDisplayName;
 	}
 
-	public addAttachToConnectionsToBeDisposed(connUri: string) {
+	public addAttachToConnectionsToBeDisposed(connUri: string, connectionProfile: ConnectionProfile) {
 		this._connectionUrisToDispose.push(connUri);
+
+		if (!this._connectionsForAttachTo.has(connUri)) {
```

Comment:
You don't want to update the connection if the URI is already there? Is adding existing URIs a common scenario?